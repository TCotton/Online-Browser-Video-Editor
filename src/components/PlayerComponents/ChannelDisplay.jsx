import React, {useState} from 'react';
import {useSelector} from "react-redux";
import {peakFrequencyLeft, peakFrequencyRight} from '../slices/audioSlice';
import D3BarChart from './D3BarChart';

const ChannelDisplay = () => {
    const rangeLeft = useSelector(peakFrequencyLeft);
    const rangeRight = useSelector(peakFrequencyRight);
    const [def, setDefault] = useState('default');

    const setDefaultClass = () => {
        if(def === 'default') setDefault('');
    }

    return (
        <>
            <div className={`left ${def}`} data-testid="ChannelDisplay">
                <D3BarChart data={rangeLeft} width={250} height={250} padding={0} colour="rgb(32, 224, 187)" setClass={setDefaultClass} />
            </div>
            <div className={`right ${def}`}>
                <D3BarChart data={rangeRight} width={250} height={250} padding={0} colour="rgb(14, 175, 163)" setClass={setDefaultClass} />
            </div>
        </>
    )
}

export default ChannelDisplay;