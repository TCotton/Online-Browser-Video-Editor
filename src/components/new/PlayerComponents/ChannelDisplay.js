import React, {useState, useCallback} from 'react';
import {useSelector} from "react-redux";
import {peakFrequency} from '../slices/audioSlice';
//import BarChart from './BarChart';
import D3BarChart from './D3BarChart';
import Circles from './AnimatedCircles';

const Foo = () => {
    const [ref, setRef] = useState(null);

    const onRefChange = useCallback(node => {
        // ref value changed to node
        setRef(node); // e.g. change ref state to trigger re-render
        if (node !== null) {
            console.log('null');
            // node is null, if DOM node of ref had been unmounted before
        } else {
            console.log('not null');
            // ref value exists
        }
    }, );

    return <h1 ref={onRefChange}>Hey</h1>;
}

const ChannelDisplay = () => {
    const range = useSelector(peakFrequency);

    return (
        <>
            <div className="left">
                <D3BarChart data={[1,2,3]} width={250} height={250} padding={2} />
                <Circles />
            </div>
            <div className="right" />
        </>
    )
}

export default ChannelDisplay;