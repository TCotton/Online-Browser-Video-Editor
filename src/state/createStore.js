import {configureStore} from '@reduxjs/toolkit';
import counterReducer from '../components/new/slices/counterSlice';
import filesReducer from '../components/new/slices/filesSlice';

export default configureStore({
    reducer: {
        counter: counterReducer,
        files:  filesReducer,
    },
});