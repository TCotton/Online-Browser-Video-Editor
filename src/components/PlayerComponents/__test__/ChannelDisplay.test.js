import React from 'react';
import ChannelDisplay from '../ChannelDisplay';
import { storeSpy, expectRedux } from 'expect-redux';
import {configureStore} from '../../../state/createStore';

xdescribe('ChannelDisplay', () => {
    let store;

    beforeEach(() => {
        store = configureStore([storeSpy]);
    })

})