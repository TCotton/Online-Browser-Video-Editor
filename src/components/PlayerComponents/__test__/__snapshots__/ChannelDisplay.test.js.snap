// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelDisplay should match snapshot 1`] = `
<body>
  <div>
    <div
      class="left "
      data-testid="ChannelDisplay"
    >
      <svg
        data-testid="svg"
        height="100%"
        preserveAspectRatio="none"
        viewBox="0 0 250 250"
        width="100%"
      >
        <g
          class="rects"
          data-testid="rects"
          transform="translate(0,0)"
        >
          <rect
            class="rect"
            fill="rgb(32, 224, 187)"
            height="982.2834645669292"
            width="250"
            y="-732.2834645669292"
          />
        </g>
      </svg>
    </div>
    <div
      class="right "
    >
      <svg
        data-testid="svg"
        height="100%"
        preserveAspectRatio="none"
        viewBox="0 0 250 250"
        width="100%"
      >
        <g
          class="rects"
          data-testid="rects"
          transform="translate(0,0)"
        >
          <rect
            class="rect"
            fill="rgb(14, 175, 163)"
            height="982.2834645669292"
            width="250"
            y="-732.2834645669292"
          />
        </g>
      </svg>
    </div>
  </div>
</body>
`;
